timely_beliefs/beliefs/classes.py

```diff
@@ -661,17 +661,26 @@ class BeliefsSeries(pd.Series):
 
     _metadata = METADATA
 
-    @property
-    def _constructor(self):
-        def f(*args, **kwargs):
-            """Pre-Pandas 2.0, call __finalize__() after construction to inherit metadata."""
-            if version.parse(pd.__version__) < version.parse("2.0.0"):
+    # Pre-Pandas 2.0, call __finalize__() after construction to inherit metadata.
+    if version.parse(pd.__version__) < version.parse("2.0.0"):
+
+        @property
+        def _constructor(self):
+            def f(*args, **kwargs):
                 return BeliefsSeries(*args, **kwargs).__finalize__(
                     self, method="inherit"
                 )
-            return BeliefsSeries(*args, **kwargs)
 
-        return f
+            return f
+
+    else:
+
+        @property
+        def _constructor(self):
+            def f(*args, **kwargs):
```

Comment:
Not sure if it provides any major benefit (except from, maybe, readability), but I think we can use the function `functools.partial`:

```python

@property
def _constructor(self):
    return functools.partial(BeliefsSeries)
```